client/extensions/woocommerce/components/sparkline/index.js

```diff
@@ -0,0 +1,111 @@
+/**
+ * External dependencies
+ */
+import React, { PropTypes, Component } from 'react';
+import classNames from 'classnames';
+import { extent as d3Extent } from 'd3-array';
+import { scaleLinear as d3ScaleLinear } from 'd3-scale';
+import { select as d3Select } from 'd3-selection';
+import { line as d3Line } from 'd3-shape';
+import ReactFauxDOM from 'react-faux-dom';
```

Comment:
I think we can get rid of `ReactFauxDOM` dependency.

If I understand correctly, it solves the problem of React not updating the `svg` when new props are introduced or changed. This is due to d3's DOM manipulation not included in React's DOM diff reconciliation process. More on React-Faux-DOM [here](https://oli.me.uk/2015/09/09/d3-within-react-the-right-way/).

Here is a clever implementation that uses a series of methods to render the svg. 
https://hackernoon.com/building-d3-components-with-react-7510e4743288

By pulling the methods out of the render function, they can be used to re-draw on `componentDidUpdate`
```js
componentDidMount() {
    this.drawSparkline();
}

componentDidUpdate() {
    this.redrawSparkline();
}
```
Worth an investigation.